ios-xcframework/fastlane/Fastfile

```diff
@@ -29,7 +28,28 @@ lane :upload_xcframework_to_s3 do
 
   # Tradeoff: We are "wasting" space on S3 by uploading the same archive under different name.
   # What we get in return is a 1-1 correspondency between archive and podspec, which keeps the setup simpler.
-  upload_xcframework_files_to_s3(archive_path: xcframework_archive_path, id: tag)
+  upload_xcframework_files_to_s3(archive_path: XCFRAMEWORK_ARCHIVE_PATH, id: tag)
+
+  # Create a draft GitHub release for the tag and upload the archive.
+  # It's a bit of a stretch to do this in the upload lane, but it's convenient.
+  # Future iterations might want a dedicated CI step that collects all binaries, not just the iOS ones.
+  next unless options.fetch(:create_github_release, true)
+
+  create_github_release
+end
+
+lane :create_github_release do
+  puts ':github: Creating GitHub release draft' if is_ci
+
+  create_release(
+    repository: 'wordpress-mobile/gutenberg-mobile',
+    version: "This is a test – id #{Time.now.to_i}",
+    # Passing nil here will result in no release description text on GitHub
+    release_notes_file_path: nil,
+    release_assets: XCFRAMEWORK_ARCHIVE_PATH.to_s,
+    # TODO: read from options, options.fetch(:beta_release, true)
+    prerelease: true
+  )
```

Comment:
@mokagio If the tag is created for a regular release, it's likely that the GitHub release is also created when this is executed. In fact, the tag is usually created when publishing the GitHub release.

I know the PR is draft but I'd like to share a couple of questions:
1. Should we also upload the podspec? In the future, it would be great if each GitHub release contains all the needed artifacts to integrate Gutenberg Mobile in the app.
2. Now that the Javascript bundle (i.e. the file `App.js`) is embedded in the XCFramework, I'm wondering if we could also upload the source maps. These files are required to symbolicate the stack traces of crashes. Currently, we are versioning them in the repository ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/bundle/ios/App.js.map)), but makes more sense to storage them somewhere else. WDYT?

Let me know if I can help you with this, I'll be more than happy to contribute 🙂!

Reply:
Thanks for looking into this @fluiddot !

> Should we also upload the podspec? In the future, it would be great if each GitHub release contains all the needed artifacts to integrate Gutenberg Mobile in the app.

We upload the `podspec` as part of the XCFramework generation. Someone downloading the XCFramework from the release artifacts manually would presumably be integrating the framework manually, too. Otherwise, all they need to do is point their `Podfile` to the version they desire.

The idea to upload the artifacts here is so that, once the `podspec` in the root of the repo is up to date with the new setup, we'll be able to use GitHub releases to store the stable binaries, which seems a bit more community / OSS friendly.

> I'm wondering if we could also upload the source maps.

Sure can do

> If the tag is created for a regular release, it's likely that the GitHub release is also created when this is executed. In fact, the tag is usually created when publishing the GitHub release.

Good point 🤔 It's been a while since I wrote this code, but I guess I was assuming a change in workflow where we'd be pushing the tag from the release manager machine and let the automation do the rest on the GitHub side.

Would that work?

I should think about it a bit more and check whether it could fit with the existing release checklist.